File: app/src/main/java/com/parabola/newtone/mvp/view/PlayerView.java
package com.parabola.newtone.mvp.view;

import com.parabola.domain.interactor.player.PlayerInteractor.RepeatMode;
import com.parabola.domain.model.Track;

import java.util.List;

import moxy.MvpView;
import moxy.viewstate.strategy.alias.AddToEndSingle;
import moxy.viewstate.strategy.alias.OneExecution;

@AddToEndSingle
public interface PlayerView extends MvpView {
    void setArtist(String artistName);
    void setAlbum(String albumTitle);
    void setTitle(String trackTitle);
    void setDurationText(String durationFormatted);
    void setDurationMs(int durationMs);
    void setIsFavourite(boolean isFavourite);

    void setPlaybackButtonAsPause();
    void setPlaybackButtonAsPlay();
    void setRepeatMode(RepeatMode repeatMode);
    void setShuffleEnabling(boolean enable);

    void setCurrentTimeMs(int currentTimeMs);

    void setTimerButtonVisibility(boolean visible);

    void setViewPagerSlide(boolean lock);

    void refreshTracks(List<Track> tracks);
    void setAlbumImagePosition(int currentTrackPosition, boolean smoothScroll);

    @OneExecution
    void showToast(String toastMessage);
}

File: app/src/main/java/com/parabola/newtone/presentation/player/PlayerView.kt
package com.parabola.newtone.presentation.player

import com.parabola.domain.interactor.player.PlayerInteractor
import com.parabola.domain.model.Track
import moxy.MvpView
import moxy.viewstate.strategy.alias.AddToEndSingle

@AddToEndSingle
interface PlayerView : MvpView {
    fun setArtist(artistName: String)
    fun setAlbum(albumTitle: String)
    fun setTitle(trackTitle: String)
    fun setDurationText(durationFormatted: String)
    fun setDurationMs(durationMs: Int)
    fun setIsFavourite(isFavourite: Boolean)

    fun setPlaybackButtonAsPause()
    fun setPlaybackButtonAsPlay()
    fun setRepeatMode(repeatMode: PlayerInteractor.RepeatMode)
    fun setShuffleEnabling(enable: Boolean)

    fun setCurrentTimeMs(currentTimeMs: Int)

    fun setTimerButtonVisibility(visible: Boolean)

    fun setViewPagerSlide(lock: Boolean)

    fun refreshTracks(tracks: List<Track>)
    fun setAlbumImagePosition(currentTrackPosition: Int, smoothScroll: Boolean)

    //в градусах
    fun setTrackSettingsRotation(rotation: Float)

    fun setRootViewOpacity(alpha: Float)
    fun setRootViewVisibility(visible: Boolean)
}
